src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,98 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import React from 'react';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  currentPage: number;
+  totalPages: number;
+  perPage: number;
+  items: string[];
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  currentPage,
+  totalPages,
+  perPage,
+  items,
+  onPageChange,
+}) => {
+  const handleArrowClick = (direction: string, condition: boolean) => {
+    const newPage = direction === 'prev'
+      ? currentPage - 1
+      : currentPage + 1;
+
+    if (!condition) {
+      onPageChange(newPage);
+    }
+  };
+
+  const pages = getNumbers(1, Math.ceil(totalPages / perPage));
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === pages.length;
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={classNames('page-item', { disabled: isFirstPage })}>
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={isFirstPage}
+            onClick={() => handleArrowClick('prev', isFirstPage)}
+          >
+            «
+          </a>
+        </li>
+        {pages.map(page => {
+          const isChosen = page === currentPage;
+
+          return (
+            <li
+              key={page}
+              className={classNames('page-item', { active: isChosen })}
+            >
+              <a
+                data-cy="pageLink"
+                className="page-link"
+                href={`#${page}`}
+                onClick={() => {
+                  onPageChange(page);
+                }}
```

Comment:
![image](https://user-images.githubusercontent.com/98764641/228186323-0d7f44c8-bb81-405a-b375-4ca1882616c8.png)

Reply:
still can`t run my tests in VS Code( If there are still any issues with my code, let me know.